Initial commit with cases. Please see descr in README.md

miledy206 opened the pull request.

miledy206 committed:
Initial commit with cases. Please see descr in README.md

AlexKitIt commented on tests/conftest.py:
```diff
@@ -0,0 +1,39 @@
+import pytest
+import string
+
+@pytest.fixture
+def generation_correct_password():
+    """Генератор простого пароля с цифрами и буквами на 13 символов"""
+    line = string.ascii_letters + string.digits
+    passw = ''
+    for i in range(12):
+        passw = passw + random.choice(line)
+    return passw
+
+@pytest.fixture
+def generation_incorrect_password():
+    """Генератор простого пароля с цифрами и буквами на 4 символов"""
+    line = string.ascii_letters + string.digits
+    passw = ''
+    for i in range(3):
+        passw = passw + random.choice(line)
+    return passw
+
+@pytest.fixture
+def generation_correct_email():
+    """Генератор email согласно маске для кейсов регистрации"""
+    login = 'safronova_olesia_12_' + str(random.randint(100, 999)) + '@ya.ru'
+    return login
+
```
Нужно исправить: все это лучше вынести в отдельный модуль helper. Это не фикстур задача

miledy206 replied:
вынесла

AlexKitIt commented on tests/conftest.py:
```diff
@@ -0,0 +1,39 @@
+    login = 'safronova_olesia_12_' + str(random.randint(100, 999)) + '@ya.ru'
+    return login
+
+@pytest.fixture
+def login_correct_email():
+    """Созданный заранее и зарегистрированный тестовый пользователь"""
+    login = 'safronova_olesya_12_123@ya.ru'
+    return login
+
+@pytest.fixture
+def login_correct_pass():
+    """Созданный заранее и зарегистрированный тестовый пользователь"""
+    passw = 'Yandex1234'
+    return passw
```
Нужно исправить: это вынести в модуль data как тестовые данные

miledy206 replied:
вынесла

AlexKitIt commented on tests/test_constructor.py:
```diff
@@ -0,0 +1,57 @@
+    def test_navigation_inside_constructor_without_login_sauce(self):
+        """Проверка перехода в подменю 'Соусы' без авторизации"""
+
+        driver = webdriver.Chrome()
+        driver.maximize_window()
```
Нужно исправить: инстанс драйвера должен подниматься и закрываться в фикстуре, не в тесте

miledy206 replied:
сделала

AlexKitIt commented on tests/test_constructor.py:
```diff
@@ -0,0 +1,57 @@
+        driver.get('https://stellarburgers.nomoreparties.site/')
+
+        WebDriverWait(driver, 3).until(
+            expected_conditions.visibility_of_element_located((By.XPATH, './/span[text() = "Булки"]')))
```
Нужно исправить: здесь и далее по всему коду: необходимо создать модуль locators в котором будут все элементы с локаторами и методом поиска. Так все это поддерживать удобнее

miledy206 replied:
вынесла

AlexKitIt commented on tests/test_constructor.py:
```diff
@@ -0,0 +1,57 @@
+
+        driver.find_element(By.XPATH, './/span[text() = "Соусы"]').click()
+
+        assert driver.find_element(By.XPATH, './/h2[text() = "Соусы"]')
```
Нужно исправить: здесь и далее во всех тестах конструктора: проверка открытого таба возможна только по visability_of_element_located, простое нахождение в DOM не даст гарантии что таб открылся

miledy206 replied:
добавила проверку на class

AlexKitIt commented on tests/test_login_page.py:
```diff
@@ -0,0 +1,136 @@
+    def test_login_correct_data_via_login_btn(self, login_correct_email, login_correct_pass):
+        """Позитивная проверка на авторизацию с помощью кнопки 'Войти в аккаунт' на главной странице"""
+
+        self.login = login_correct_email
+        self.password = login_correct_pass
```
Нужно исправить: тестовые классы создаются только для объединения тестовых методов по логической или функциональной общности, ну иногда для скоупа фикстур. Но своих методов и атрибутов они не имеют.
Если видишь необходимость в этих атрибутах - это должны быть просто переменные

miledy206 replied:
исправила

AlexKitIt commented on tests/test_login_page.py:
```diff
@@ -0,0 +1,136 @@
+
+        check_value = driver.find_element(By.XPATH, './/label[text() = "Имя"]/parent::div/input').get_attribute(
+            'value')
+        assert check_value == 'Олеся'
```
Можно лучше: если добавишь assert message будет проще понять что именно и где у нас упало

AlexKitIt commented on tests/test_registration_page.py:
```diff
@@ -0,0 +1,67 @@
+        WebDriverWait(driver, 3).until(expected_conditions.visibility_of_element_located(
+            (By.XPATH, './/h2[text() = "Вход"]')))
+
+        assert driver.find_element(By.XPATH, './/button[text() = "Войти"]')
```
Нужно исправить: наличие кнопки Войти не говорит что регистрация прошла успешно. Попробуй поискать вариант понадежнее

miledy206 replied:
добавила успешный логин после реги

miledy206 committed:
Rework after review from Alex:

create 3 additional files:
- data.py - login, pass value for test user
- helper.py - generation login and pass for reg
- locators.py - all using locators in proj

add 2 new fixture:
- run Chrome driver
- success login

add an addit. verifications:
- for reg - success login
- for constructor - check class value for selected element

replace JS script for clicking btn to internal func click()

AlexKitIt commented on tests/conftest.py:
```diff
@@ -0,0 +1,32 @@
+from selenium.webdriver.support import expected_conditions
+from selenium.webdriver.support.wait import WebDriverWait
+
+from tests.locators import *
+from tests.data import *
```
Можно лучше: импорт через * чреват скорым появлением цикличных импортов

AlexKitIt commented on tests/conftest.py:
```diff
@@ -0,0 +1,32 @@
+def success_login():
+    driver = webdriver.Chrome()
+    driver.maximize_window()
+    driver.get('https://stellarburgers.nomoreparties.site/')
```
Нужно исправить: не, так нельзя. Инстанс получаем только одной фикстурой. В твоей реализации нужно было в фикстуру success_login передать run_driver и далее
 WebDriverWait(run_driver , 3).until(expected_conditions.visibility_of_element_located(
        ButtonsLocators.FIND_LOGIN_BTN_MAIN_PAGE)).click() и т.д.

AlexKitIt commented on tests/conftest.py:
```diff
@@ -0,0 +1,32 @@
+    driver.find_element(*PersonalDataLocators.FIND_PASSWORD_FIELD).send_keys(password)
+
+    driver.find_element(*ButtonsLocators.FIND_LOGIN_BTN_LOGIN_PAGE).click()
+    yield driver
+    driver.quit()
```
Вот этого соответственно тоже тут быть не должно

miledy206 replied:
Я тогда не понимаю, как нужно передавать фикстуру в фикстуру. Можно, пожалуйста, пример. Особенно с выходом. Данный пример я нашла на stackoverflow, примеров фикстур, когда объект одного передаётся в фикстуру другого я не нашла :(

AlexKitIt commented on tests/data.py:
```diff
@@ -0,0 +1,4 @@
+"""Созданный заранее и зарегистрированный тестовый пользователь"""
+email = 'safronova_olesya_12_123@ya.ru'
```
Можно лучше: это все лучше хранить в отдельном классе, чтобы его импортировать было удобнее

AlexKitIt commented on tests/helper.py:
Лайк, но только он не относится к директории test, он должен лежать в корне, как и data, как и locators

miledy206 committed:
Rework after review from Alex:

move 3 files to the main directory: data.py, helper.py, locators.py

remove 1 fixture:
- success login
because I don't understand how I can rework it, so I could run it successfully

AlexKitIt commented on tests/test_constructor.py:
```diff
@@ -0,0 +1,51 @@
+    def test_navigation_inside_constructor_without_login_sauce(self, run_driver):
+        """Проверка перехода в подменю 'Соусы' без авторизации"""
+
+        driver = run_driver
```
Если сразу назовешь фиксутру driver то и эта строка нигде не понадобится)

AlexKitIt commented on tests/test_login_page.py:
```diff
@@ -0,0 +1,111 @@
+from selenium.webdriver.support import expected_conditions
+from selenium.webdriver.support.wait import WebDriverWait
+
+from locators import *
+from data import *
```
А импорты почему опять через *? Не привыкай к этому - потом очень много проблем будет с цикличными импортами. Положи в классы
